### PR TITLE
feat: Added upgrade-python-requirements workflow

### PR DESCRIPTION
**Pull request by Jawayria**
JIRA: https://openedx.atlassian.net/browse/BOM-2602

### CODE REVIEW

**Review by mraarif — APPROVED** [review]

File: .github/workflows/upgrade-python-requirements.yml
Comment (sarina) [inline .github/workflows/upgrade-python-requirements.yml:60]: We have removed this on our repos, relying on CODEOWNERS. Please see https://github.com/edx/xblock-lti-consumer/blob/master/.github/workflows/upgrade-python-requirements.yml

**Review by sarina — CHANGES_REQUESTED** [review]

File: .github/workflows/upgrade-python-requirements.yml
Comment (stvstnfrd) [reply .github/workflows/upgrade-python-requirements.yml]: @sarina I don't think this is the same thing.
- `CODEOWNERS` tags folks when PRs are _successfully_ created
- This build step (that's since been removed) notifies folks when this upgrade job _fails_ to create a PR in the first place.

I don't think we want to remove this notification. If the jenkins job were to start failing, we wouldn't know; we'd just stop getting upgrade PRs.
Given that there could be some weeks w/out upgrades, I wouldn't want us to rely on human memory and pattern recognition to remember/notice that it's been too long since we've gotten upgrades on a particular repo.

**Review by stvstnfrd — CHANGES_REQUESTED** [review]

**Comment by sarina** [comment]
@Jawayria I retract my comment, @stvstnfrd is correct here.

**Comment by Jawayria** [comment]
> @Jawayria I retract my comment, @stvstnfrd is correct here.

Okay

File: .github/workflows/upgrade-python-requirements.yml
Comment (sarina) [inline .github/workflows/upgrade-python-requirements.yml:67]: this should be community-engineering, no?

Comment (Jawayria) [reply .github/workflows/upgrade-python-requirements.yml:67]: Can you please mention team's complete email address so that I could update here?

Comment (sarina) [reply .github/workflows/upgrade-python-requirements.yml:67]: same as the other one. community-engineering@edx.opsgenie.net

**Review by sarina — APPROVED** [review]
Please squash commits before merging

**Comment by Jawayria** [comment]
@stvstnfrd please review

**Pull request merged**
